build/dnsmasq/build.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/bash
+#
+# {{{ CDDL HEADER
+#
+# This file and its contents are supplied under the terms of the
+# Common Development and Distribution License ("CDDL"), version 1.0.
+# You may only use this file in accordance with the terms of version
+# 1.0 of the CDDL.
+#
+# A full copy of the text of the CDDL should have accompanied this
+# source. A copy of the CDDL is also available via the Internet at
+# http://www.illumos.org/license/CDDL.
+# }}}
+
+# Copyright 2022 OmniOS Community Edition (OmniOSce) Association.
+
+. ../../lib/build.sh
+
+PROG=dnsmasq
+VER=2.86
+PKG=ooce/network/dnsmasq
+SUMMARY="Lightweight, easy to configure DNS forwarder"
+DESC="dnsmasq is a lightweight, easy to configure DNS forwarder, designed to "
+DESC+="provide DNS (and optionally DHCP and TFTP) services to a small-scale network."
+
+set_arch 64
+set_mirror 'https://thekelleys.org.uk/'
+set_checksum sha256 '28d52cfc9e2004ac4f85274f52b32e1647b4dbc9761b82e7de1e41c49907eb08'
+
+BASEDIR=$PREFIX/$PROG
+CONFFILE=/etc$BASEDIR/$PROG.conf
+EXECFILE=$PREFIX/sbin/dnsmasq
+
+copy_sample_config() {
+    local relative_conffile=${CONFFILE#/}
+    local dest_confdir=$DESTDIR/${relative_conffile%/*}
+
+    logmsg "-- copying sample config"
+    logcmd mkdir -p "$dest_confdir" || logerr "mkdir failed"
+    logcmd cp $TMPDIR/$BUILDDIR/$PROG.conf.example $DESTDIR/$relative_conffile \
+        || logerr "copying configs failed"
+}
+
+# No configure
+configure64() { :; }
+
+XFORM_ARGS="
+    -DPREFIX=${PREFIX#/}
+    -DBASEDIR=${BASEDIR#/}
+    -DEXECFILE=$EXECFILE
+    -DCONFFILE=$CONFFILE
+    -DUSER=$PROG
+    -DGROUP=$PROG
+    -DPROG=$PROG
+"
+
+MAKE_ARGS_WS="
+    CC=$CC
+    CFLAGS=\"-DNO_IPSET $CFLAGS $CFLAGS64 $CTF_CFLAGS\"
```

Comment:
This package will be built for a range of different OmniOS versions, and not all of those build packages with CTF, so you only want to conditionally include CTF_CFLAGS here.
The best way to deal with this is to wrap the assignments for `MAKE_ARGS_WS` and `MAKE_INSTALL_ARGS_WS` into the `configure64()` function, and drop `$CTF_CFLAGS`.
For those versions where CTF is the default, `CFLAGS` will be automatically extended with `CTF_CFLAGS` in the `build()` function which is what ends up calling configure64().
https://github.com/omniosorg/omnios-extra/blob/master/build/lz4/build.sh is an example of this.